test/sample.jl

```diff
@@ -226,6 +226,27 @@
         @test !ismissing(chain[1].a)
     end
 
+    @testset "Thin chain by a factor of `thinning`" begin
+        K, N = 3, 100  # (thinning factor, number of samples)
+
+        # Run a thinned chain with N samples thinned by factor of K.
+        Random.seed!(1234)
+        thinned_chain = sample(MyModel(), MySampler(), N; sleepy = true, thinning = K)
+        @test length(thinned_chain) == N
+        @test ismissing(thinned_chain[1].a)
+
+        # Run a long chain with N * K samples.
+        Random.seed!(1234)
+        long_chain = sample(MyModel(), MySampler(), N * K; sleepy = true)
+        @test length(long_chain) == N * K
+        @test ismissing(long_chain[1].a)
+
+        # Check that thinned_chain is every k-th sample in long_chain.
+        # NOTE: `i` begins at 2 since first sample in `thinned_chain` is `missing`.
+        @test all([thinned_chain[i].a == long_chain[(i-1) * K + 1].a for i in 2:N])
```

Comment:
I'm a bit picky here but I think it would be nice to use a more descriptive name such as `thinning` instead of `K`. We also don't have to test the reference chain, we just compare with it:
```suggestion
        # Run a thinned chain with `N` samples thinned by factor of `thinning`.
        Random.seed!(1234)
        N = 100
        thinning = 3
        chain = sample(MyModel(), MySampler(), N; sleepy = true, thinning = thinning)
        @test length(chain) == N
        @test ismissing(chain[1].a)

        # Repeat sampling without thinning.
        Random.seed!(1234)
        ref_chain = sample(MyModel(), MySampler(), N * thinning; sleepy = true)
        @test all(chain[i].a === ref_chain[(i - 1) * thinning + 1].a for i in 1:N)
```

Reply:
I haven't a strong opinion here. I'll just note that "thinning" is a rather common term for Bayesian modelers.

As for alternate names, STAN uses `thin`, tensorflow-probability uses `num_steps_between_results`. Do you have a preference or another term in mind.

Reply:
Sorry, I misread again. I guess you were referring to `K` in the test only? Yes, let me change that.